core/app/[locale]/(default)/product/[slug]/_components/wishlist-button/index.tsx

```diff
@@ -0,0 +1,154 @@
+import { removeEdgesAndNodes } from '@bigcommerce/catalyst-client';
+import { getTranslations } from 'next-intl/server';
+import { cache } from 'react';
+
+import { Streamable } from '@/vibes/soul/lib/streamable';
+import { Favorite } from '@/vibes/soul/primitives/favorite';
+import { getSessionCustomerAccessToken, isLoggedIn } from '~/auth';
+import { client } from '~/client';
+import { graphql } from '~/client/graphql';
+import { revalidate } from '~/client/revalidate-target';
+import { TAGS } from '~/client/tags';
+import { ExistingResultType } from '~/client/util';
+
+import { WishlistButtonDropdown } from './dropdown';
+
+const wishlistButtonLimit = 50;
+const WishlistButtonQuery = graphql(`
+  query WishlistButtonQuery($first: Int, $productId: Int!) {
+    customer {
+      wishlistsContainingProduct: wishlists(
+        first: $first
+        filters: { productEntityIds: [$productId] }
+      ) {
+        edges {
+          node {
+            entityId
+            name
+            items(first: 50, filters: { productEntityIds: [$productId] }) {
+              edges {
+                node {
+                  entityId
+                  productEntityId
+                  variantEntityId
+                  product {
+                    sku
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+      wishlists(first: $first) {
+        edges {
+          node {
+            entityId
+            name
+          }
+        }
+      }
+    }
+  }
+`);
+
+const getWishlistButtonData = cache(async (productId: number, customerAccessToken?: string) => {
+  if (!customerAccessToken) {
+    return null;
+  }
+
+  const { data } = await client.fetch({
+    document: WishlistButtonQuery,
+    variables: { productId, first: wishlistButtonLimit },
+    customerAccessToken,
+    fetchOptions: { next: { revalidate, tags: [TAGS.customer] } },
+  });
+
+  return data.customer;
+});
+
+export interface WishlistButtonWishlistInfo {
+  entityId: number;
+  name: string;
+  wishlistItemId?: number;
+}
+
+interface WishlistButton {
+  isProductInWishlist: boolean;
+  wishlists: WishlistButtonWishlistInfo[];
+}
+
+async function getWishlistButton(
+  t: ExistingResultType<typeof getTranslations<'Product.ProductDetails.WishlistButton'>>,
```

Comment:
You're already awaiting for data, might as well await for translations inside the function too.

Reply:
Good call